runtime/include/module.h

```diff
@@ -186,6 +220,41 @@ module_release(struct module *module)
 	return;
 }
 
+/**
+ * Check whether a module is a paid module
+ * @param module module
+ * @returns true if the module is paid, false otherwise
+ */
+static inline uint64_t
+module_is_paid(struct module *module)
+{
+	return module->replenishment_period > 0;
+}
+
+/**
+ * Get Timeout priority for Priority Queue ordering
+ * @param element module_timeout
+ * @returns the priority of the module _timeout element
+ */
+static inline uint64_t
+module_timeout_get_priority(void *element)
+{
+	return ((struct module_timeout *)element)->timeout;
+}
+
+/**
+ * Compute the next timeout given a module's replenishment period
+ * @param m_replenishment_period
+ * @return given module's next timeout
+ */
+static inline uint64_t
+get_next_timeout_of_module(uint64_t m_replenishment_period)
+{
+	uint64_t now = __getcycles();
+	return runtime_boot_timestamp
+	       + ((now - runtime_boot_timestamp) / m_replenishment_period + 1) * m_replenishment_period;
```

Comment:
I suspect that this might be better if the initial timestamp represents when the tenant/module is actually up and able to receive requests. I believe that `runtime_boot_timestamp` occurs quite a bit earlier than that before the JSON is even parsed. Not really a big deal I suppose. Just makes the initial period a sort of random duration.